### PR TITLE
Skip RemoveRedundantTableScanPredicate if predicate doesn't change

### PR DESCRIPTION
**Pull request by sopel39**
Fixes https://github.com/trinodb/trino/issues/10532

### CODE REVIEW

**Review by martint — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java [hunk 1]
```diff
@@ -203,19 +203,19 @@ public void testSubsumePartitionPartOfAFilter()
         // Join is important because it triggers PredicatePushDown logic (EffectivePredicateExtractor)
         assertDistributedPlan(
                 "SELECT l.str_col, r.str_col FROM table_int_partitioned l JOIN table_unpartitioned r ON l.int_part = r.int_col " +
-                        "WHERE l.int_part BETWEEN 2 AND 4 AND l.str_col != 'three'",
+                        "WHERE l.int_part IN (1, 4) AND l.str_col != 'three'",
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/optimizer/TestHivePlans.java:206]: Why change in an existing test?
Can we rather have an expplicit regression test for https://github.com/trinodb/trino/issues/10532 ?

File: core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java [hunk 2]
```diff
@@ -137,20 +111,25 @@ private PlanNode removeRedundantTableScanPredicate(
...
         Map<ColumnHandle, Symbol> assignments = ImmutableBiMap.copyOf(node.getAssignments()).inverse();
         Expression resultingPredicate = createResultingPredicate(
                 plannerContext,
-                session,
-                symbolAllocator,
+                context.getSession(),
+                context.getSymbolAllocator(),
                 typeAnalyzer,
-                new DomainTranslator(plannerContext).toPredicate(session, unenforcedDomain.transformKeys(assignments::get)),
+                new DomainTranslator(plannerContext).toPredicate(context.getSession(), unenforcedDomain.transformKeys(assignments::get)),
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java:125]: nit: assigning `context.getSession()` to a variable would make for a smaller diff and arguably more readable code

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java [hunk 3]
```diff
@@ -871,6 +871,7 @@ public PlanOptimizers(
                 ImmutableSet.<Rule<?>>builder()
                         .addAll(simplifyOptimizerRules) // Should be always run after PredicatePushDown
                         .add(new PushPredicateIntoTableScan(plannerContext, typeAnalyzer))
+                        .add(new RemoveRedundantTableScanPredicate(plannerContext, typeAnalyzer))
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java:874]: This is the only change in "Fix redundant predicate above table scan not being removed" commit.
i think commit boundaries were blurred during cherry of #10549

File: core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java [hunk 4]
```diff
@@ -57,7 +64,9 @@
 
     private static final Pattern<FilterNode> PATTERN =
             filter().with(source().matching(
-                    tableScan().capturedAs(TABLE_SCAN)));
+                    tableScan().capturedAs(TABLE_SCAN)
+                            // avoid extra computations if table scan doesn't have any enforced predicate
+                            .matching(node -> !node.getEnforcedConstraint().isAll())));
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java:69]: now that you cherried #10549, can you also apply the rename mentioned https://github.com/trinodb/trino/pull/10549/files#r783053560? thanks

**Review by findepi — COMMENTED** [review]
let me re-review after commit boundaries

**Pull request merged**